lib/lua/mark_eats/musicutil.lua

```diff
@@ -107,91 +107,110 @@ function MusicUtil.generate_scale(root_num, scale_type, octaves)
   return output
 end
 
---- Snap MIDI note numbers to an array of note numbers.
--- @param note_nums MIDI note number (0-127) or array of note numbers.
--- @param snap_array Array of note numbers to snap to, must be in low to high order.
--- @return Array of adjusted note numbers or a single note number.
-function MusicUtil.snap_notes_to_array(note_nums, snap_array)
-  if not note_nums or not snap_array then return nil end
-  if type(note_nums) == "number" then note_nums = {note_nums} end
-  
-  local delta
-  local prev_delta
-  local note_nums_len = #note_nums
+
+--- Snap a MIDI note number to the nearest note number in an array.
+-- @param note_num MIDI note number input (0-127).
+-- @param snap_array Array of MIDI note numbers to snap to, must be in low to high order.
+-- @return Adjusted note number.
+function MusicUtil.snap_note_to_array(note_num, snap_array)
   local snap_array_len = #snap_array
-  for n = 1, note_nums_len do
-    if snap_array_len == 1 then
-      note_nums[n] = snap_array[1]
-    elseif note_nums[n] >= snap_array[snap_array_len] then
-      note_nums[n] = snap_array[snap_array_len]
-    else
-      prev_delta = math.huge
-      for s = 1, snap_array_len + 1 do
-        if s > snap_array_len then
-          note_nums[n] = note_nums[n] + prev_delta
-          break
-        end
-        delta = snap_array[s] - note_nums[n]
-        if delta == 0 then
-          break
-        elseif math.abs(delta) >= math.abs(prev_delta) then
-          note_nums[n] = note_nums[n] + prev_delta
-          break
-        end
-        prev_delta = delta
-      end
-    end
-  end
-  if note_nums_len == 1 then return note_nums[1]
-  else return note_nums end
+	if snap_array_len == 1 then
+	  note_num = snap_array[1]
+	elseif note_num >= snap_array[snap_array_len] then
+	  note_num = snap_array[snap_array_len]
+	else
+    local delta
+    local prev_delta = math.huge
+	  for s = 1, snap_array_len + 1 do
+	    if s > snap_array_len then
+	      note_num = note_num + prev_delta
+	      break
+	    end
+	    delta = snap_array[s] - note_num
+	    if delta == 0 then
+	      break
+	    elseif math.abs(delta) >= math.abs(prev_delta) then
+	      note_num = note_num + prev_delta
+	      break
+	    end
+	    prev_delta = delta
+	  end
+	end
+
+  return note_num
+end
+
+--- Snap an array of MIDI note numbers to an array of note numbers.
+-- @param note_nums_array Array of input MIDI note numbers.
+-- @param snap_array Array of MIDI note numbers to snap to, must be in low to high order.
+-- @return Array of adjusted note numbers.
+function MusicUtil.snap_notes_to_array(note_nums_array, snap_array)
+	for i = 1, #note_nums_array do
```

Comment:
similarly, this looks mega-indented.

(we should really iron-out a formatter solution)

are you using a maiden or working locally?

Reply:
ah yes sorry i messed it up in sublime

Reply:
out of curiosity, do you use a prettifier / formatter in sublime?

Reply:
I didn't research it much, just installed this package but looking at that link it seems pretty unmaintained! https://github.com/rorydriscoll/LuaSublime